### PR TITLE
Add IAMAuthPolicy doc

### PR DESCRIPTION
**Pull request by zijun726911**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/reference/iam-auth-policy.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+               <li>&ldquo;Accepted&rdquo;</li>
+               <li>&ldquo;Ready&rdquo;</li>
+            </ul>
+         </td>
+      </tr>
+   </tbody>
+</table>
+
+
+## IAMAauthPolicy Example
```
Comment (mikhail-aws) [inline docs/reference/iam-auth-policy.md:177]: please move example above spec

File: docs/reference/iam-auth-policy.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                            "vpc-lattice-svcs:RequestHeader/header1": "value1"
+                        }
+                    }
+                }
+            ]
+        }
+```
+
+If you create the above IAMAuthPolicy in the k8s cluster, the `my-route` (and it's corresponding VPC Lattice service) will be attached with the given IAM auth policy. 
+Only HTTP traffic with header `header1:value1` will be allowed to access the `my-route`. Please check the [VPC Lattice documentation](https://docs.aws.amazon.com/vpc-lattice/latest/ug/auth-policies.html#auth-policies-common-elements) to get more detail on how lattice auth policy work.
```
Comment (mikhail-aws) [inline docs/reference/iam-auth-policy.md:209]: Didnt get what the point of this part.

**Review by mikhail-aws — CHANGES_REQUESTED** [review]

**Review by mikhail-aws — APPROVED** [review]
feel free  to merge **after** addressing comments

File: docs/reference/iam-auth-policy.md (hunk 2 above)
Comment (zijun726911) [reply docs/reference/iam-auth-policy.md:211]: Do you mean remove the whole `## IAMAauthPolicy Example` section?

**Pull request merged**